### ISSUE TITLE
Noting non-bundled plugins in the plugin docs

### ISSUE THREAD

**Issue by acchen97** [elastic/logstash#7829]
In the past, we had a banner in the plugin docs that said something along the lines of "This plugin is not installed by default, install it with bin/logstash-plugin install ..."

We should add this back to mitigate usability confusion.

/cc @dedemorton @jordansissel @suyograo

**Comment by dedemorton** [comment]
@acchen97  The reason the current docs are confusing with regards to what needs to be installed is that they are missing the Installation section that should be included with any community plugins. Like the sections that shows up here: https://www.elastic.co/guide/en/logstash/5.2/plugins-codecs-cef.html

This is an error in the docgen script. Here you’ll see that we have some conditional coding in the asciidoc to show the Installation instructions when the value of {default_plugin} is 0. https://raw.githubusercontent.com/elastic/logstash/master/docs/include/plugin_header.asciidoc

The generated asciidoc file for each plugin that is NOT bundled with Logstash needs to have `:default_plugin: 0` at the top of the file in order to get the installation section to display. 

For example, the asciidoc attributes at the top of the generated asciidoc for the cef codec docs should say:
```
:plugin: cef
:type: codec
:default_plugin: 0
```

I've already communicated the fix to development.

**Comment by dedemorton** [comment]
@acchen97 If we need a quick fix, someone could go through the generated files in logstash-docs and add `:default_plugin: 0` to the top of any files that require it.

**Comment by jordansissel** [comment]
Working on it.

**Comment by jordansissel** [comment]
`:default_plugin: 0` for only non-default plugins?

**Comment by jordansissel** [comment]
Code written: here's a sample of what `master` branch docs would look like:

https://github.com/elastic/logstash-docs/compare/sample-docgen-master?expand=1

**Comment by jordansissel** [comment]
5.5 branch: https://github.com/elastic/logstash-docs/compare/5.5...sample-docgen-5.5?

**Comment by jordansissel** [comment]
@dedemorton I'm seeing lots of this on 5.5:

```diff
-[id="plugins-{type}s-{plugin}"]
+[id="plugins-{type}-{plugin}"]
```

did we hand-fix something on logstash-docs and not in the upstream repos?

**Comment by dedemorton** [comment]
We've fixed the plugin docgen script to handle this problem, so I am closing this issue. The 5.6 docs might have some plugins that contain installation instructions, but shouldn't. This will be fixed when we regenerate the plugin docs for 5.6. I've created this issue to track the work: https://github.com/elastic/logstash/issues/8571.

Closing this issue because the details are no longer valid.
